react-app/src/utils/leaguehelpers.js

```diff
@@ -33,51 +33,107 @@ export default {
 
 	getDate: (league, dateId) => {
 		//dateId: league.scheduledMatches.dateId
-		if(league == null || dateId === ''){
+		var getdate = {};
+
+		if(league === null || dateId === ''){
 			return null;
 		}
 
 		league.dates.forEach((date) => {
 			if(date.id === dateId){
-				return date;
+				getdate = date;
 			}
 		});
+
+		//return leagues.dates.filter((date) => {date.id === dateId});
+
+		return getdate;
 	},
 
 	getNumInLeague: (league, teamId) => {
 		//teamId: league.scheduledMatches.teamOneId
-		if(league === null || dateId === ''){
+		var teamNum = '';
+
+		if(league === null || teamId === ''){
 			return '';
 		}
 
 		league.teams.forEach((team) => {
 			if(team.id === teamId){
-				return team.numInLeague;
+				teamNum = team.numInLeague;
 			}
 		});
 
+		//return leagues.teams.filter((team) => {team.id === teamId});
+
+		return teamNum;
 	},
 
 	getVenueName: (lookups, fieldId) => {
```

Comment:
This helper method isn't necessary since the list of venues is keyed by their id's in the store